src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -24,6 +24,7 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 			}
 			BaseType = baseType;
 			Interfaces = interfaces ?? new System.Type[0];
+			_uniqueInterfaces = new HashSet<System.Type>(Interfaces);
```

Comment:
Ok,  no it can not, as hashCode is calculated using just xor (which is strange as there will be lots of collisions)